File: javapatterns/src/main/java/org/iut/nantes/facade/cours/Videoprojecteur.java
package org.iut.nantes.facade.cours;

public class Videoprojecteur {
    public void brancherVideoprojecteur(Multiprises multiprises) {
        System.out.println("Vidéoprojecteur branché.");
    }

    public void débrancherVideoprojecteur(Multiprises multiprises) {
        System.out.println("Vidéoprojecteur débranché.");
    }

    public void relierOrdinateur(Ordinateur ordinateur) {
        System.out.println("Ordinateur relié.");
    }

    public void allumerVidéoprojecteur() {
        System.out.println("Vidéoprojecteur allumé.");
    }

    public void éteindreVidéoprojecteur() {
        System.out.println("Vidéoprojecteur éteint.");
    }
}

File: kotlinpatterns/src/main/kotlin/org/iut/nantes/facade/cours/Videoprojecteur.kt
package org.iut.nantes.facade.cours

class Videoprojecteur {
    fun brancherVideoprojecteur(multiprises: Multiprises?) {
        println("Vidéoprojecteur branché.")
    }

    fun débrancherVideoprojecteur(multiprises: Multiprises?) {
        println("Vidéoprojecteur débranché.")
    }

    fun relierOrdinateur(ordinateur: Ordinateur?) {
        println("Ordinateur relié.")
    }

    fun allumerVidéoprojecteur() {
        println("Vidéoprojecteur allumé.")
    }

    fun éteindreVidéoprojecteur() {
        println("Vidéoprojecteur éteint.")
    }
}